### ISSUE TITLE
twisted.test.test_udp calls reactor.iterate()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11293]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1201|
|Type|defect|
|Created|2005-09-19 18:03:36Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1201 1201
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1127153016000000 1127153016000000
changetime__1167098487000000 1167098487000000
version__ 
owner__ 
cc__exarkun cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
The receiving end of most of these tests should probably expose a Deferred-based
interface for dealing with results.

The ones near connectUDP may need a notification mechanism so they know when to
proceed.  Does one exist (is startProtocol called at the right time)?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Fixed in test_udp-reactor.iterate-1201

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_4"></a>|@jerub set owner to @exarkun|
|-|-|

Be careful when merging this branch. If import changes are made in the [#2154](https://github.com/twisted/twisted/issues/2154) branch, twisted/test/test_udp.py will be in conflict in the first hunk. As these are import problems please be very careful when merging, but don't feel compelled to merge forward.

Thankyou so much for cleaning up the warnings here. I hate those things.

There is still a reactor.iterate() call somewhere in the Multicast stuff. testInterface and testJoinLeave call runUntilSuccess. Please investigate and remove the need for reactor.iterate().

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @jerub|
|-|-|

Thanks.  Missed those, since on Linux the loop body never executes so the deprecation warning is never emitted.

I fixed the remaining tests and deleted runUntilSuccess.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_7"></a>|@jerub set owner to @exarkun|
|-|-|

Much better. Please merge this branch now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [19114]) Merge test_udp-reactor.iterate-1201

Author: exarkun
Reviewer: jerub
Fixes [#11293](https://github.com/twisted/twisted/issues/11293)

Rewrite UDP tests which explicitly iterate the reactor so that they use
Deferreds and callbacks and so forth instead.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
